### PR TITLE
Add `verify` feature into blockchain tests

### PR DESCRIPTION
**Pull request by rajarshimaitra**
### Description
As pointed out by @LLFourn https://discord.com/channels/753336465005608961/753367451319926827/919895212434784316 our verification checks are not fully being CI tested. We should check for transaction verification in our sync process. This is already written in `wallet.sync()` but is feature gated by `verify` flag. 
https://github.com/bitcoindevkit/bdk/blob/64e88f0e006c68315142d53dc35b633327dde4b5/src/wallet/mod.rs#L1540-L1555
Thus running the `blockchain_tests` with just `verify` feature would be enough to check the verification process for wallet transactions.

For non wallet transaction verification we should probably include new blockchain tests.

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

### CODE REVIEW

**Comment by rajarshimaitra** [comment]
Rebased.

File: .github/workflows/cont_integration.yml [hunk 1]
```diff
@@ -89,13 +89,13 @@ jobs:
       matrix:
         blockchain:
           - name: electrum
-            features: test-electrum
+            features: test-electrum,verify
           - name: rpc
-            features: test-rpc
+            features: test-rpc,verify
```
Comment (afilini) [inline .github/workflows/cont_integration.yml]: This is not necessary as rpc doesn't verify transactions

Comment (rajarshimaitra) [reply .github/workflows/cont_integration.yml]: Right, missed that..

**Comment by rajarshimaitra** [comment]
There is a test failure in the sync verification at wallet module. Which wasn't checked previously. 

This part of the code is already refactored away in #502  and the error doesn't occur there. So I am just porting this PR's changes into 502 and closing this one.

**Pull request closed without merging**